### PR TITLE
스웨거 추가 및 보안

### PR DESCRIPTION
**Pull request by purr6**

### CODE REVIEW

File: src/auth/auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { jwtConstants } from './constants';
+import { Request } from 'express';
```
Comment (gwjang) [inline src/auth/auth.guard.ts:9]: express가 아니라 fastify로 수정해주세요

File: src/auth/auth.guard.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+@Injectable()
+export class AuthGuard implements CanActivate {
```
[the remark this answers is not in the document]
Comment (purr6) [reply src/auth/auth.guard.ts]: [79df2fb](https://github.com/gwjang/cat-shop/pull/2/commits/79df2fb4d6c3275dc3c33bb0b8dcf6f5ebe8dc16)

AuthGuard 사용했어요.

File: src/auth/constants.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const jwtConstants = {
+  secret: '1234',
```
[the remark this answers is not in the document]
Comment (purr6) [reply src/auth/constants.ts]: [79df2fb](https://github.com/gwjang/cat-shop/pull/2/commits/79df2fb4d6c3275dc3c33bb0b8dcf6f5ebe8dc16)

env로 빼고 랜덤 생성해서 넣었어요.

File: src/auth/Roles.guard.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+//   }
+//
+//   // eslint-disable-next-line @typescript-eslint/no-unused-vars
+//   private matchRoles(roles: any, roles2: any) {
+//     return true;
+//   }
+// }
+
+@Injectable()
+export class RolesGuard implements CanActivate {
```
[the remark this answers is not in the document]
Comment (purr6) [reply src/auth/Roles.guard.ts]: [79df2fb](https://github.com/gwjang/cat-shop/pull/2/commits/79df2fb4d6c3275dc3c33bb0b8dcf6f5ebe8dc16)

뺐어요.

File: src/cat/cats.controller.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import { CreateCatDto } from './dto/create-cat.dto';
+import { CatsService } from './cats.service';
+import { Roles } from '../auth/Roles.Decorator';
+
+@Controller('cats')
+export class CatsController {
+  constructor(private catsService: CatsService) {}
+
+  @Post()
+  @Roles(['admin'])
```
[the remark this answers is not in the document]
Comment (purr6) [reply src/cat/cats.controller.ts]: [79df2fb](https://github.com/gwjang/cat-shop/pull/2/commits/79df2fb4d6c3275dc3c33bb0b8dcf6f5ebe8dc16)

뺐어요.

File: src/auth/auth.guard.ts (hunk 1 above)
Comment (purr6) [reply src/auth/auth.guard.ts]: [79df2fb](https://github.com/gwjang/cat-shop/pull/2/commits/79df2fb4d6c3275dc3c33bb0b8dcf6f5ebe8dc16)

이제 사용하지 않아요.

**Review by gwjang — APPROVED** [review]

**Pull request merged**